### PR TITLE
HCAP-1125 | Use uniform date library on client & server

### PR DESCRIPTION
**Pull request by taraepp**
https://freshworks.atlassian.net/browse/HCAP-1125
- replaced moment with dayjs in client packages
- dayjs/moment (or other date libraries) are a dependency of the MUI Pickers Utils Provider, used their [installation instructions](https://material-ui-pickers.dev/getting-started/installation)
- the instructions said to use an older version than what is used for the server, which is also older than what's available to install, want to verify that we *must* use the older version
- TODO: seek and destroy instances of manual date parsing

### CODE REVIEW

**Review by fwpushan — DISMISSED** [review]
LGTM

**Review by fwpushan — DISMISSED** [review]
LGTM

File: client/package.json [hunk 1]
```diff
@@ -12,22 +12,22 @@
...
-    "@date-io/moment": "1.3.13",
+    "@date-io/dayjs": "^1.3.13",
     "@material-ui/core": "4.11.0",
     "@material-ui/icons": "4.9.1",
     "@material-ui/lab": "4.0.0-alpha.49",
     "@material-ui/pickers": "3.2.10",
     "@react-keycloak/web": "2.1.4",
     "classnames": "2.2.6",
     "cross-fetch": "3.0.6",
+    "dayjs": "^1.11.1",
```
Comment (alva-fresh) [inline client/package.json:23]: Can we lock this dependency?

File: client/package.json [hunk 2]
```diff
@@ -12,22 +12,22 @@
     "eject": "react-scripts eject"
   },
   "dependencies": {
-    "@date-io/moment": "1.3.13",
+    "@date-io/dayjs": "^1.3.13",
```
Comment (alva-fresh) [inline client/package.json:15]: And this as well

File: server/package-lock.json [hunk 3]
```diff
@@ -1045,6 +1045,20 @@
         "node": "^12.13.0 || ^14.15.0 || ^16.10.0 || >=17.0.0"
       }
     },
+    "node_modules/@jest/globals": {
```
Comment (alva-fresh) [inline server/package-lock.json:1048]: I don't think we need to introduce changes to `server/package-lock` or root `package-lock`. I have a PR to address these

File: package-lock.json [hunk 4]
```diff
@@ -452,9 +453,9 @@
       }
     },
     "node_modules/async": {
-      "version": "3.2.0",
-      "resolved": "https://registry.npmjs.org/async/-/async-3.2.0.tgz",
-      "integrity": "sha512-TR2mEZFVOj2pLStYxLht7TyfuRzaydfpxr3k9RpHIzMgw7A64dzsdqCxH1WJyQdoe8T10nDXd9wnEigmiuHIZw==",
+      "version": "3.2.3",
```
Comment (alva-fresh) [inline package-lock.json:456]: see my comment below

**Review by alva-fresh — APPROVED** [review]

**Pull request merged**